### PR TITLE
feat: remove provider, revert modified recoveryParam

### PR DESCRIPTION
**Pull request by ieow**
remove provider
#123

revert modified recoveryParam
#122

### CODE REVIEW

File: tests/login.spec.ts [hunk 1]
```diff
@@ -107,8 +107,8 @@ variable.forEach((testVariable) => {
       // get key details
       await checkLogin(coreKitInstance);
 
-      const result = await coreKitInstance.provider.request({ method: "eth_chainId", params: [] });
-      assert.strictEqual(result, DEFAULT_CHAIN_CONFIG.chainId);
+      // const result = await coreKitInstance.provider.request({ method: "eth_chainId", params: [] });
+      // assert.strictEqual(result, DEFAULT_CHAIN_CONFIG.chainId);
```
Comment (matthiasgeihs) [inline tests/login.spec.ts:111]: wanna remove these later?

File: tests/login.spec.ts [hunk 2]
```diff
@@ -137,16 +137,16 @@ variable.forEach((testVariable) => {
...
+      //   throw new Error("should not reach here");
+      // } catch (error) {}
 
       // setup provider
-      await coreKitInstanceWithoutProvider.setupProvider({ chainConfig: DEFAULT_CHAIN_CONFIG });
-      const result = await coreKitInstanceWithoutProvider.provider.request({ method: "eth_chainId", params: [] });
-      assert.strictEqual(result, DEFAULT_CHAIN_CONFIG.chainId);
+      // await coreKitInstanceWithoutProvider.setupProvider({ chainConfig: DEFAULT_CHAIN_CONFIG });
+      // const result = await coreKitInstanceWithoutProvider.provider.request({ method: "eth_chainId", params: [] });
+      // assert.strictEqual(result, DEFAULT_CHAIN_CONFIG.chainId);
```
Comment (matthiasgeihs) [inline tests/login.spec.ts:149]: remove?

**Review by matthiasgeihs — COMMENTED** [review]
Remove `public async setupProvider`?

Remove dependencies?
- @web3auth-mpc/ethereum-provider
- @web3auth/base-provider
- @metamask/swappable-obj-proxy

**Review by himanshuchawla009 — CHANGES_REQUESTED** [review]

File: src/mpcCoreKit.ts [hunk 3]
```diff
@@ -762,13 +749,10 @@ export class Web3AuthMPCCoreKit implements ICoreKit {
       throw err;
     });
 
-    let { r, s, recoveryParam } = await client.sign(tss, Buffer.from(msgHash).toString("base64"), true, "", "keccak256", {
+    const { r, s, recoveryParam } = await client.sign(tss, Buffer.from(msgHash).toString("base64"), true, "", "keccak256", {
       signatures,
     });
 
-    if (recoveryParam < 27) {
```
[the remark this answers is not in the document]
Comment (ieow) [reply src/mpcCoreKit.ts:769]: yes.
for pcs , they add another check to revert the modified value

[the remark this answers is not in the document]
Comment (himanshuchawla009) [reply src/mpcCoreKit.ts:769]: got it, will ask devrels to validate and write a example around it using viem

**Review by himanshuchawla009 — APPROVED** [review]

**Comment by himanshuchawla009** [comment]
need to resolve merge conflicts

File: demo/redirect-flow-example/src/App.tsx [hunk 4]
```diff
@@ -409,20 +423,44 @@ function App() {
...
+    //   blockExplorerUrl: "https://opbnbscan.com",
+    //   ticker: "BNB",
+    //   tickerName: "opBNB",
+    // };
+    // if (coreKitInstance.status === COREKIT_STATUS.LOGGED_IN) {
+    //   let localProvider = new EthereumSigningProvider({ config: { chainConfig: newChainConfig } });
+    //   localProvider.setupProvider(coreKitInstance);
+    //   setProvider(localProvider);
+    // }
+    // or
```
Comment (matthiasgeihs) [inline demo/redirect-flow-example/src/App.tsx:442]: Let's try to remove all commented out code before merging. Doesn't look good in production code...

File: demo/redirect-flow-example/src/App.tsx [hunk 5]
```diff
@@ -399,8 +405,16 @@ function App() {
...
+    if (coreKitInstance.status === COREKIT_STATUS.LOGGED_IN) {
+      let localProvider = new EthereumSigningProvider({ config: { chainConfig: newChainConfig } });
+      localProvider.setupProvider(coreKitInstance);
+      setProvider(localProvider);
+    }
+    // or
+    // provider.sendAsync({
+    //   method: "wallet_switchEthereumChain",
+    //   params: [{ chainId: newChainConfig.chainId }],
+    // })
```
Comment (matthiasgeihs) [inline demo/redirect-flow-example/src/App.tsx:417]: Let's try to remove all commented out code before merging. Doesn't look good in production code...

File: demo/redirect-flow-example/src/App.tsx [hunk 6]
```diff
@@ -378,8 +375,17 @@ function App() {
...
+    if (coreKitInstance.status === COREKIT_STATUS.LOGGED_IN) {
+      let localProvider = new EthereumSigningProvider({ config: { chainConfig: newChainConfig } });
+      localProvider.setupProvider(coreKitInstance);
+      setProvider(localProvider);
+    }
+    // or 
+    // provider.sendAsync({
+    //   method: "wallet_switchEthereumChain",
+    //   params: [{ chainId: newChainConfig.chainId }],
+    // })
```
Comment (matthiasgeihs) [inline demo/redirect-flow-example/src/App.tsx:388]: Let's try to remove all commented out code before merging. Doesn't look good in production code...

**Comment by matthiasgeihs** [comment]
something that i wondered just now:
the demos heavily rely on usage of the provider, right? how do you deal with this? @ieow do you simply remove all of the respective functionality from the demos?

**Review by himanshuchawla009 — APPROVED** [review]
lgtm

**Pull request merged**